docs/algorithms/ijon/algorithm_en.md

```diff
@@ -0,0 +1,45 @@
+# IJON
+
+## What is IJON?
+
+IJON is an annotation mechanism that allows PUTs to return new types of feedback to fuzzers, and a fuzzer that supports those feedbacks, proposed by SysSec. Many famous fuzzers are classified as coverage-guided fuzzers, which try to find new behaviors in a program by receiving code coverage as feedback from PUTs. A typical coverage-guided fuzzer has the following weaknesses: 
+
+- They do not care about the order in which the code coverage was obtained. For example, suppose there is a bug whose triggering condition is "function B must be executed immediately after function A is executed". Because the fuzzer cannot distinguish between the input that causes the bug and the input that causes the execution of function A after the execution of function B, it may try only the latter and overlook the bug. More to the point, many algorithms cannot distinguish between "input that causes both function A and function B to be called" and "a set of input that causes function A to be called and input that causes function B to be called" in the first place, and if one of the two is already tested, the other won't be tested.
```

Comment:
the orderにin whichは必須だと思うけどどうかな（the number of timesとかとは話が違う気がする？）
executeはあんまり自動詞で使うイメージないけどどうですか？（runなら分かる）
"the input that causes both"は"an input ~"の方が良さそう
two inputsとしてしまうと、"one of them"のthemがtwo inputsなのか"the input ~ and two inputs"なのか不明っぽくて、setにしたい。あるいはone of themの表現何か別の良いやつないですかね

Reply:
全体的に日本語をそのまま直したっぽい受動的な文が多くて読みづらかったので、可能な限り主語を明確にしたいと思っています。

> the orderにin whichは必須だと思う

では"the order in which it obtained the code coverage"が妥当かと思います。

> executeはあんまり自動詞で使うイメージない

'whose triggering condition is "executing function B immediately after the execution of function A."'はどうでしょうか。

> two inputsとしてしまうと...

同じことを思いましたが、"a set of A and B"という表現を聞いたことがない上、それぞれが長いので読みづらかったです。"If the two inputs are tested first, ..."とかはどうでしょうか。

Reply:
あっSuggestionこれかあ
after the execution以外は全部賛同したい

```suggestion
- It does not care about the order in which it obtained the code coverage. For example, suppose there is a bug whose triggering condition is "executing function B immediately after executing function A". Because the fuzzer cannot distinguish between the input that causes the bug and the one that causes the execution of function A after function B, it may try only the latter and overlook the bug. More to the point, many algorithms cannot distinguish between "the input that causes both function A and function B to be called" and "two inputs such that one of them executes only function A and the other executes only function B." If the latter two inputs are tested first, the fuzzer will not test the former one.
```

Reply:
LGTM